phoenix-core/src/main/java/org/apache/phoenix/util/SchemaUtil.java

```diff
@@ -897,4 +941,86 @@ public static boolean hasRowTimestampColumn(PTable table) {
+    }
+
+    public static PName getPhysicalHBaseTableName(PName pName, boolean isNamespaceMapped, PTableType type) {
+        return getPhysicalHBaseTableName(pName.toString(), isNamespaceMapped, type);
+    }
+
+    public static PName getPhysicalHBaseTableName(byte[] tableName, boolean isNamespaceMapped, PTableType type) {
+        return getPhysicalHBaseTableName(Bytes.toString(tableName), isNamespaceMapped, type);
+    }
+
+    public static TableName getPhysicalTableName(String fullTableName, ReadOnlyProps readOnlyProps) {
+        return getPhysicalName(Bytes.toBytes(fullTableName), readOnlyProps);
+    }
+
+    public static TableName getPhysicalTableName(byte[] fullTableName, Configuration conf) {
+        return getPhysicalTableName(fullTableName, isNamespaceMappingEnabled(
+                isSystemTable(fullTableName) ? PTableType.SYSTEM : null, new ReadOnlyProps(conf.iterator())));
+    }
+
+    public static TableName getPhysicalName(byte[] fullTableName, ReadOnlyProps readOnlyProps) {
+        return getPhysicalTableName(fullTableName,
+                isNamespaceMappingEnabled(isSystemTable(fullTableName) ? PTableType.SYSTEM : null, readOnlyProps));
+    }
+
+    public static TableName getPhysicalTableName(byte[] fullTableName, boolean isNamespaceMappingEnabled) {
+        if (!isNamespaceMappingEnabled) { return TableName.valueOf(fullTableName); }
+        String tableName = getTableNameFromFullName(fullTableName);
+        String schemaName = getSchemaNameFromFullName(fullTableName);
+        return TableName.valueOf(schemaName, tableName);
+    }
+
+    public static String getSchemaNameFromHBaseFullName(byte[] tableName, ReadOnlyProps props) {
+        if (tableName == null) { return null; }
+        int index = isNamespaceMappingEnabled(null, props) ? indexOf(tableName, QueryConstants.NAMESPACE_SEPARATOR_BYTE)
+                : indexOf(tableName, QueryConstants.NAME_SEPARATOR_BYTE);
+        if (index < 0) { return StringUtil.EMPTY_STRING; }
+        return Bytes.toString(tableName, 0, index);
+    }
+
+    public static PName getPhysicalHBaseTableName(String tableName, boolean isNamespaceMapped, PTableType type) {
+        if (!isNamespaceMapped) { return PNameFactory.newName(tableName); }
+        return PNameFactory
+                .newName(tableName.replace(QueryConstants.NAME_SEPARATOR, QueryConstants.NAMESPACE_SEPARATOR));
+    }
+
+    public static boolean isSchemaCheckRequired(PTableType tableType, ReadOnlyProps props) {
+        if (PTableType.TABLE.equals(tableType) && isNamespaceMappingEnabled(tableType, props)) { return true; }
+        return false;
+    }
+
+    public static boolean isNamespaceMappingEnabled(PTableType type, ReadOnlyProps readOnlyProps) {
+        return readOnlyProps.getBoolean(QueryServices.IS_NAMESPACE_MAPPING_ENABLED,
+                QueryServicesOptions.DEFAULT_IS_NAMESPACE_MAPPING_ENABLED)
+                && (type == null || !PTableType.SYSTEM.equals(type)
```

Comment:
Can the type be null here. If not, I would annotate the argument as @Nonnull and have a Preconditions.checkNotNull(type) check at the beginning of the method.